src/encoded/static/components/submission/submission-view.js

```diff
@@ -534,8 +533,8 @@ export default class SubmissionView extends React.Component{
             newHierarchy = modifyHierarchy(hierarchy, keyIdx, parentKeyIdx);
             validCopy[keyIdx] = 1; // new object has no incomplete children yet
             validCopy[parentKeyIdx] = 0; // parent is now not ready for validation
-            typesCopy[keyIdx] = type;
         }
+        typesCopy[keyIdx] = type;
```

Comment:
Important-est front-end edit. Everything else is minor and unimportant (extraState can be reverted).